experimental/dds/tree2/src/feature-libraries/editable-tree-2/lazyField.ts

```diff
@@ -224,12 +223,6 @@ export abstract class LazyField<TKind extends FieldKind, TTypes extends AllowedT
 	}
 }
 
-function assertIsLazySequence<TTypesSource extends AllowedTypes>(
```

Comment:
I pulled this inline since it was only used in one place, and the type guard assertion was actually making the type less correct, not more correct.